### PR TITLE
Removed quotas cap

### PR DESCRIPTION
**Pull request by cheina97**
# Description

This PR removes the cap on CPU and memory for tenants.

@QcFe can you confirm if it is secure? If I remember well a normal user cannot customize it's tenant.

### CODE REVIEW

**Comment by frisso** [comment]
Wouldn't be possible to set those quota parameters as properties of the tenant itself? With "zero" as "no limit"?

**Comment by cheina97** [comment]
> Wouldn't be possible to set those quota parameters as properties of the tenant itself? With "zero" as "no limit"?

They should be added in the tenant spec. This means a change in the Crownlabs API. I think it should be a good improvement, but it could require some times.

**Comment by cheina97** [comment]
> Would you mind setting the new values inside the helm chart too? Setting as default the old values that were hardcoded...

Of course 👍

**Comment by cheina97** [comment]
@QcFe PR is ready or review

File: operators/deploy/tenant-operator/values.yaml [hunk 1]
```diff
@@ -19,6 +19,14 @@ configurations:
   maxConcurrentReconciles: 1
   sandboxClusterRole: crownlabs-sandbox
   tenantNamespaceKeepAlive: 168h
+  tenant:
+    resourcecaps:
+      # Maximum number of virtual cpus that can be requested by a tenant.
```
Comment (frisso) [inline operators/deploy/tenant-operator/values.yaml:24]: I would write here whether those limits are *per instance* or *per tenant*.
If it is *per tenant*, those numbers look pretty small.

**Comment by frisso** [comment]
If we want to give resources to our students that exceed their capabilities at home, I feel that, per each instance, students should be allowed to create up to 4 cores and 16GB RAM at least (personally, I would say even 8 and 32).
Max 10 instances, instead, looks fair to me.
@QcFe @cheina97 what do you think?

File: operators/deploy/tenant-operator/values.yaml (hunk 1 above)
Comment (cheina97) [reply operators/deploy/tenant-operator/values.yaml:24]: I think @QcFe is going to override those values with ArgoCD or through CrownOps. But we can surely increase the default value. I forgot to add in the description that 0 means that the it no cap in resources, I'm going to add it.

File: operators/pkg/forge/resourcequota.go [hunk 2]
```diff
@@ -25,17 +25,17 @@ import (
...
-
-	// CapInstance -> The cap number of instances that can be requested by a Tenant.
-	CapInstance = 10
 )
 
 var (
+	// CapInstance -> The cap number of instances that can be requested by a Tenant.
+	CapInstance int
+
 	// CapCPU -> The cap amount of CPU cores that can be requested by a Tenant.
```
Comment (frisso) [inline operators/pkg/forge/resourcequota.go:34]:
```suggestion
	// CapCPU -> The total amount of CPU cores that can be requested by a Tenant.
```

**Comment by frisso** [comment]
> > If we want to give resources to our students that exceed their capabilities at home, I feel that, per each instance, students should be allowed to create up to 4 cores and 16GB RAM at least (personally, I would say even 8 and 32). Max 10 instances, instead, looks fair to me. @QcFe @cheina97 what do you think?
> 
> At the moment the caps are 25 for the CPU, 50GB for RAM and 10 for the instances. What do you think about putting a huge amount of resources as cap? Something like 150 CPU, 300GB of RAM, and 50 instances? Remember that this limit is for all tenants (not only students). The limits of student's resources should be managed with workspaces.

Current limits (25 CPU, 50GB, 10 instances) look perfect to me.

**Comment by frisso** [comment]
LGTM, for me we can merge.

**Pull request merged**